UnitySDK/Assets/ML-Agents/Scripts/Policy/BarracudaPolicy.cs

```diff
@@ -25,17 +26,20 @@ public class BarracudaPolicy : IPolicy
         /// Sensor shapes for the associated Agents. All Agents must have the same shapes for their Sensors.
         /// </summary>
         List<int[]> m_SensorShapes;
+        IRewardProvider m_RewardProvider;
 
         /// <inheritdoc />
         public BarracudaPolicy(
             BrainParameters brainParameters,
             NNModel model,
-            InferenceDevice inferenceDevice)
+            InferenceDevice inferenceDevice,
+            IRewardProvider rewardProvider)
```

Comment:
Does the barracuda policy need the reward provider? Why?

Reply:
you're right.